cmd/bugzilla-mass-cloner/main.go

```diff
@@ -0,0 +1,120 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/test-infra/prow/bugzilla"
+	prowflagutil "k8s.io/test-infra/prow/flagutil"
+)
+
+type options struct {
+	prowflagutil.BugzillaOptions
+
+	fromRelease string
+	toRelease   string
+
+	dryRun bool
+}
+
+func gatherOptions() options {
+	o := options{}
+	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+
+	fs.BoolVar(&o.dryRun, "dry-run", true, "Dry run for testing. Uses API tokens but does not mutate.")
+	fs.StringVar(&o.fromRelease, "from-release", "", "From which targeted release the bug will be cloned to")
+	fs.StringVar(&o.toRelease, "to-release", "", "To which release value the cloned bugs will hold")
+
+	o.BugzillaOptions.AddFlags(fs)
+
+	if err := fs.Parse(os.Args[1:]); err != nil {
+		logrus.WithError(err).Fatal("could not parse input")
+	}
+
+	return o
+}
+
+func (o *options) validate() error {
+	if err := o.BugzillaOptions.Validate(o.dryRun); err != nil {
+		return err
+	}
+
+	if o.fromRelease == "" {
+		return fmt.Errorf("--from-release must be specified")
+	}
+
+	if o.toRelease == "" {
+		return fmt.Errorf("--from-release must be specified")
+	}
+
+	if o.fromRelease == o.toRelease {
+		return fmt.Errorf("--from-release and --to-release can't hold the same value")
+	}
+
+	return nil
+}
+
+type bugzillaClient interface {
+	SearchBugs(filters map[string]string) ([]*bugzilla.Bug, error)
+	CloneBugWithMutate(bug *bugzilla.Bug, mutations []func(bug *bugzilla.BugCreate)) (int, error)
+}
+
+func (o options) getBugsByStatus(client bugzillaClient, statuses []string) ([]*bugzilla.Bug, error) {
+	var bugsToClone []*bugzilla.Bug
+
+	for _, status := range statuses {
+		bugs, err := client.SearchBugs(map[string]string{"target_release": o.fromRelease, "status": status})
+		if err != nil {
+			return nil, fmt.Errorf("couldn't search for bugs: %v", err)
+		}
+		bugsToClone = append(bugsToClone, bugs...)
+	}
+	return bugsToClone, nil
+}
+
+func (o options) massCloneBugs(client bugzillaClient, bugs []*bugzilla.Bug) error {
+	var errs []error
+	for _, bug := range bugs {
+		if o.dryRun {
+			logrus.WithField("bug-id", bug.ID).Info("(dry-run) Will clone bug...")
+			continue
+		}
+
+		addTarget := []func(bug *bugzilla.BugCreate){func(bug *bugzilla.BugCreate) { bug.TargetRelease = []string{o.toRelease} }}
+		newBugID, err := client.CloneBugWithMutate(bug, addTarget)
+		if err != nil {
+			errs = append(errs, fmt.Errorf("couldn't clone bug with id %d: %v", bug.ID, err))
+		}
```

Comment:
nit: this can be unconditional, `NewAggregate` handles `nil`s AFAIK

Reply:
It does, but I would like to have a clean error message with the bug ID included in the error for better logging.